mobly/controllers/android_device_lib/adb.py

```diff
@@ -121,27 +139,27 @@ def _exec_cmd(self, args, shell, timeout):
             AdbError: The adb command exit code is not 0.
             AdbTimeoutError: The adb command timed out.
         """
+        cmd_str = ' '.join(args)
         proc = subprocess.Popen(
             args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=shell)
         process = psutil.Process(proc.pid)
+        if timeout and timeout <= 0:
+            raise Error('Timeout is not a positive value: %s' % timeout)
         if timeout and timeout > 0:
             try:
                 process.wait(timeout=timeout)
             except psutil.TimeoutExpired:
                 process.terminate()
-                raise AdbTimeoutError('Timed out Adb cmd "%s". timeout: %s' %
-                                      (args, timeout))
-        elif timeout and timeout < 0:
-            raise AdbTimeoutError('Timeout is a negative value: %s' % timeout)
+                raise AdbTimeoutError(cmd=cmd_str, timeout=timeout)
 
         (out, err) = proc.communicate()
         ret = proc.returncode
-        logging.debug('cmd: %s, stdout: %s, stderr: %s, ret: %s', args, out,
+        logging.debug('cmd: %s, stdout: %s, stderr: %s, ret: %s', cmd_str, out,
                       err, ret)
         if ret == 0:
             return out
         else:
-            raise AdbError(cmd=args, stdout=out, stderr=err, ret_code=ret)
+            raise AdbError(cmd=cmd_str, stdout=out, stderr=err, ret_code=ret)
```

Comment:
The `cmd` member should stay as it was - a list. Changing it to a string breaks API without a real need. I would suggest only changing the way that the string representation of the exception works (or not changing it)

Reply:
Done